src/materialized/bin/materialized.rs

```diff
@@ -161,6 +161,7 @@ fn run() -> Result<(), failure::Error> {
     };
 
     let data_directory = popts.opt_get_default("data-directory", PathBuf::from("mzdata"))?;
+    fs::create_dir_all(&data_directory)?;
```

Comment:
Don't feel obligated, because there's nothing worse than getting punished for a good deed, but this would be a great place to add some additional context:

```rust
use failure::ResultExt;
// ...
fs::create_dir_all(&data_directory).with_context(|e| format!("creating data directory: {}", e))?`
```

Reply:
oh 100% totally forgot about this discussion